src/pipeline/incrementer_adder.rs

```diff
@@ -0,0 +1,57 @@
+use futures::{ready, Stream};
+use pin_project::pin_project;
+use std::ops::{Add, AddAssign};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// implementation of how `incrementer_adder` works
+/// holds the stream being consumed, and a stateful incrementer that increases with every item
+#[pin_project]
+pub struct IncrementerAdder<St: Stream> {
```

Comment:
should we hide these types behind a `#[cfg(test)]` attribute? They don't seem to be useful for actual work we'll be doing for IPA and probably serve just one purpose to show the idea how to work with streams, so we may as well hide them from global namespace